src/gui/sdlmain.cpp

```diff
@@ -744,6 +746,19 @@ static void log_display_properties(const int in_x,
 	        out_x, out_y, out_par);
 }
 
+static SDL_Point get_initial_window_position_or_default(int default_val)
+{
+	int x, y;
+	if (sdl.desktop.window.initial_x_pos >= 0 &&
+	    sdl.desktop.window.initial_y_pos >= 0) {
+		x = sdl.desktop.window.initial_x_pos;
+		y = sdl.desktop.window.initial_y_pos;
+	} else {
+		x = y = default_val;
+	}
+	return {x, y};
+}
+
```

Comment:
Minor comment - the DOSBox-stype code uses "immediate action"  logic.

When it comes to return value handling, this means that there can be multiple return points in a function.  

Contrast this with 100% object-oriented code base where all objects maintain a valid state and "throw" on invalid states: this means all the corner-case and error handling is moved out-of-band into the `catch` handlers.  In these cases, the sunny-day functions don't have to have error-state handling in them - and this makes that code very amenable to a single-return point coding styles.  (and in those cases, I also prefer a single return point).

DOSBox's overall program flow is very C-based - passing around low-level types, pointers to data and such. So this means many functions have to perform lots of checking.  This fit with the behaviour of DOS itself, where port and values literally requires entirely different code-flows.  

So the DOSBox authors went with "immediate-action" coding style - meaning multiple return points are possible - and it tends to suit the code base.

I'm bringing up just to give you context (in this very simple and tiny example).  We're trying to adhere to this style (when it makes sense) - so as to fit with the overall readability.

So in this case, allowing for multiple return points would produce something like:

```c++
static SDL_Point get_initial_window_position_or_default(int default_val)
{

	const auto &w = sdl.desktop.window;
	if (w.initial_x_pos >= 0 && w.initial_y_pos >= 0)
		return {w.initial_x_pos, w.initial_y_pos}

	assert(default_val > 0);
	return {default_val, default_val};
}
```

Just a very minor comment!  

I also added the assertion that we'd expect the default to be positive?